app/models/pet.rb

```diff
@@ -2,12 +2,26 @@ class Pet < ApplicationRecord
   validates :name, presence: true
   validates :age, presence: true, numericality: true
   belongs_to :shelter
+  has_many :application_pets
+  has_many :applications, through: :application_pets
 
   def shelter_name
     shelter.name
   end
 
+  def self.find_pet(id)
+    find(id)
```

Comment:
Same idea here -- no need to encapsulate this in a model method, it's very conventional to keep it in the controller!